sycl/source/detail/graph_impl.cpp

```diff
@@ -1639,7 +1639,7 @@ node modifiable_command_graph::addImpl(dynamic_command_group &DynCGF,
 
   graph_impl::WriteLock Lock(impl->MMutex);
   std::shared_ptr<detail::node_impl> NodeImpl = impl->add(DynCGFImpl, DepImpls);
-  return sycl::detail::createSyclObjFromImpl<node>(NodeImpl);
+  return sycl::detail::createSyclObjFromImpl<node>(std::move(NodeImpl));
```

Comment:
Is this of any benefit if `createSyclObjFromImpl()` doesn't accept r-value reference?

Reply:
A right, I guess it removes one of the copies - will call a move constructor of shared_ptr.